shared/src/main/scala/io/kaitai/struct/languages/AwkwardCompiler.scala

```diff
@@ -803,6 +823,14 @@ class AwkwardCompiler(
             s"${if (isRepeat) ".content()" else ""}" +
             s"${if (unionIndex.contains("child")) ".content<" + unionIndex.split("_").last + ">()" else ""}" + 
             s");" else ""}"
+        case enumType: EnumType =>
+          var builderName = idToStr(id)
+          val enumMapClass = builderName+"s_t_map"
```

Comment:
@ManasviGoyal not sure what is the reliable way of doing this naming operation

Reply:
@zonca The reason `scdms.ksy` was giving errors because `StringBuilder` was not added for instances with `enumType`.
I have made a few changes for the above and also added the indexedoption for the instances. Can you give me access to add these changes to this branch?

The changes work. However, the issue now is that in `IndexedOptionBuilder`, when we do `append_valid`, it adds the index starting 0 but the added string can have any index associated with it. I am not sure how to add the strings in a way the index matches with string key.

Reply:
I selected "Allow edits by maintainers" so you should be able to push to my branch.

Reply:
I tried but it still says permission denied

Reply:
the easiest would be if you make a pull request in my repository to zonca:enum_indexedbuilder,
once we merge that, commits are going to show up here.